tripy/nvtripy/backend/api/executable.py

```diff
@@ -78,6 +79,24 @@ def __init__(
         Stores metadata, like shapes and data types, for each input to the executable.
         """
 
+        # Build accessor map from compile-time access plans
+        self._accessor_map: Dict[str, callable] = {}
+        name_to_index = {name: idx for idx, name in enumerate(self._arg_names)}
+
+        def make_accessor(arg_index: int, steps: Tuple[Union[str, int], ...]):
+            def accessor(inputs, idx=arg_index, stps=steps):
+                v = inputs[idx]
+                for s in stps:
```

Comment:
Can we just capture arg_index/steps from the enclosing scope?
```suggestion
            def accessor(inputs):
                v = inputs[arg_index]
                for s in steps:
```